containers-tests/benchmarks/Map.hs

```diff
@@ -84,6 +86,9 @@ main = do
         , bench "mapMaybeWithKey" $ whnf (M.mapMaybeWithKey (const maybeDel)) m
         , bench "lookupIndex" $ whnf (lookupIndex keys) m
         , bench "union" $ whnf (M.union m_even) m_odd
+        , bench "union_identical" $ whnf (M.union m_even) m_even
+        , bench "union_sparse" $ whnf (M.union m_even) m_sparse
+        , bench "union_into_sparse" $ whnf (M.union m_sparse) m_even
```

Comment:
Two more with `m_sparse` and `m_odd` would show if there's a slowdown in `insertR`.